examples/smooth.py

```diff
@@ -0,0 +1,311 @@
+    space="x",
+    time_range=None,
+):
+    # If no time range is specified, plot the entire time series
+    if time_range is None:
+        time_range = slice(0, ds_raw.time[-1])
+
+    selection = {
+        "time": time_range,
+        "individuals": individual,
+        "keypoints": keypoint,
+        "space": space,
+    }
+
+    fig, ax = plt.subplots(2, 1, figsize=(10, 6))
+
+    for ds, color, label in zip(
+        [ds_raw, ds_smooth], ["k", "r"], ["raw", "smooth"]
+    ):
+        # plot position time series
+        pos = ds.position.sel(**selection)
+        ax[0].plot(
+            pos.time,
+            pos,
+            color=color,
+            lw=2,
+            alpha=0.7,
+            label=f"{label} {space}",
+        )
+
+        # generate interpolated dataset to avoid NaNs in the PSD calculation
+        ds_interp = interpolate_over_time(ds, max_gap=None, print_report=False)
+        pos_interp = ds_interp.position.sel(**selection)
+        # compute and plot the PSD
+        freq, psd = welch(pos_interp, fs=ds.fps, nperseg=256)
+        ax[1].semilogy(
+            freq,
+            psd,
+            color=color,
+            lw=2,
+            alpha=0.7,
+            label=f"{label} {space}",
+        )
+
+    ax[0].set_ylabel(f"{space} position (px)")
+    ax[0].set_xlabel("Time (s)")
+    ax[0].set_title("Time Domain")
+    ax[0].legend()
+
+    ax[1].set_ylabel("PSD (dB/Hz)")
+    ax[1].set_xlabel("Frequency (Hz)")
+    ax[1].set_title("Frequency Domain")
+    ax[1].legend()
+
+    plt.tight_layout()
+    fig.show()
+
+
+# %%
+# Smoothing with a median filter
+# ------------------------------
+# Here we use the :py:func:`movement.filtering.median_filter` function to
+# apply a rolling window median filter to the wasp dataset.
+# The ``window_length`` parameter is defined in seconds (according to the
+# ``time_unit`` dataset attribute).
+
+ds_wasp_medfilt = median_filter(ds_wasp, window_length=0.1)
+
+# %%
+# We see from the printed report that the dataset has no missing values
+# neither before nor after smoothing. Let's visualise the effects of the
+# median filter in the time and frequency domains.
+
+plot_raw_and_smooth_timeseries_and_psd(
+    ds_wasp, ds_wasp_medfilt, keypoint="stinger"
+)
+
+# %%
+# We see that the median filter has removed the "spikes" present around the
+# 14 second mark in the raw data. However, it has not dealt the big shift
+# occurring during the final second. In the frequency domain, we can see that
+# the filter has reduced the power in the high frequencies, without
+# affecting the very low frequency components.
+#
+# This illustrates what the median filter is good at: removing brief "spikes"
+# (e.g. a keypoint abruptly jumping to a different location for a frame or two)
+# and high frequency "jitter" (often present due to pose estimation
+# working on a per-frame basis).
+
+# %%
+# Choosing parameters for the median filter
+# -----------------------------------------
```

Comment:
```suggestion
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Maybe make this a subsection of the median filter?

Reply:
cool, I'd never used this lower-level header in rst before